scheduler/src/api/jobs/create_job.rs

```diff
@@ -182,7 +183,11 @@ async fn get_file_range_for_file(url: &str) -> Result<(u64, u64), ApiResponse<()
     Ok((start_range, end_range))
 }
 
-async fn create_sub_job(state: &Arc<AppState>, job: &Job) -> Result<SubJob, ApiResponse<()>> {
+async fn create_sub_job(
+    state: &Arc<AppState>,
+    job: &Job,
+    details: serde_json::Value,
```

Comment:
I'd strongly prefer to enforce strong typing here. Create some SubJobDetails type and derive Serialize/Deserialize for it and serialize to json when writing to db and deserialize immediately when reading from the db.

Reply:
Here are some examples that may or may not make this easier: https://docs.rs/sqlx/latest/sqlx/types/struct.Json.html